tests/abi/tools/revng-ensure-rft-equivalence/Main.cpp

```diff
@@ -0,0 +1,119 @@
+                                           cl::desc("<right input model>"),
+                                           cl::value_desc("right model"));
+
+static cl::opt<std::string> OutputFilename("o",
+                                           cl::cat(ThisToolCategory),
+                                           cl::init("-"),
+                                           llvm::cl::desc("Override "
+                                                          "output "
+                                                          "filename"),
+                                           llvm::cl::value_desc("filename"));
+
+int main(int Argc, char *Argv[]) {
+  cl::HideUnrelatedOptions({ &ThisToolCategory });
+  cl::ParseCommandLineOptions(Argc, Argv);
+
+  ExitOnError ExitOnError;
+
+  auto LeftModel = ModelInModule::load(LeftModelPath);
+  if (not LeftModel)
+    ExitOnError(LeftModel.takeError());
+
+  auto RightModel = ModelInModule::load(RightModelPath);
+  if (not RightModel)
+    ExitOnError(RightModel.takeError());
+
+  std::error_code EC;
+  llvm::ToolOutputFile OutputFile(OutputFilename,
+                                  EC,
+                                  sys::fs::OpenFlags::OF_Text);
+  if (EC)
+    ExitOnError(llvm::createStringError(EC, EC.message()));
+
+  // Gather potentially problematic functions.
+  struct FunctionPair {
+    model::RawFunctionType *Left;
+    model::RawFunctionType *Right;
+  };
+  std::map<decltype(model::Type::ID), FunctionPair> Functions;
+  for (model::UpcastableType &LeftType : LeftModel->Model->Types)
+    if (auto *Left = llvm::dyn_cast<model::RawFunctionType>(LeftType.get()))
+      Functions[Left->ID].Left = Left;
+  for (model::UpcastableType &RightType : RightModel->Model->Types)
+    if (auto *Right = llvm::dyn_cast<model::RawFunctionType>(RightType.get()))
+      Functions[Right->ID].Right = Right;
+
+  // Ensure their stack arguments have the same ID.
+  std::map<model::TypePath, model::TypePath> Replacements;
+  for (auto [_, Pair] : Functions) {
+    auto [Left, Right] = Pair;
+
+    model::Type *LeftStackArguments = Left->StackArgumentsType.get();
+    model::Type *RightStackArguments = Right->StackArgumentsType.get();
+
+    // Ignore function pairs without stack arguments.
+    revng_assert(!LeftStackArguments == !RightStackArguments);
+    if (LeftStackArguments == nullptr)
+      continue;
+
+    if (LeftStackArguments->ID != RightStackArguments->ID) {
+      model::TypePath FromPath = Right->StackArgumentsType;
+
+      RightModel->Model->Types.erase(LeftStackArguments->key());
+      auto *Struct = llvm::dyn_cast<model::StructType>(RightStackArguments);
+      revng_assert(Struct != nullptr);
+      auto Copy = model::UpcastableType::make<model::StructType>(*Struct);
+      Copy->ID = LeftStackArguments->ID;
+      auto ToPath = RightModel->Model->recordNewType(std::move(Copy));
+
+      Replacements.emplace(FromPath, ToPath);
+    }
```

Comment:
So you're overriding `StackArgumentsType` on the A model with the one from `B` model, purge duplicates and then `diff`?
Please comment.